src/modules/CurrencyModule.cpp

```diff
@@ -915,20 +917,123 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
 			return;
 		}
 
+		QString thiefName = UmikoBot::Instance().GetName(channel.guildId(), authorId);
+		QString victimName = UmikoBot::Instance().GetName(channel.guildId(), victimId);
+
+		auto& victimCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), victimId)];
+		auto& authorCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)];
+
+		EventModule* eventModule = static_cast<EventModule*>(UmikoBot::Instance().GetModuleByName("event"));
+		auto& eventConfig = eventModule->getEventServerData(channel.guildId());
+
+		if (eventConfig.eventHighRiskHighRewardRunning)
+		{
+			double successChance = (config.highRiskRewardStealSuccessChance / 100.0) * qExp(-0.0001 * qPow(amountToSteal, 1.5));
+
+			QString thiefName = UmikoBot::Instance().GetName(channel.guildId(), authorId);
+			QString victimName = UmikoBot::Instance().GetName(channel.guildId(), victimId);
+
+			std::random_device device;
+			std::mt19937 prng{ device() };
+			std::discrete_distribution<> distribution{ { 1 - successChance, successChance } };
+			int roll = distribution(prng);
+			qDebug() << "local success chance"<< successChance;
+			qDebug() << "steal success chance"<< config.stealSuccessChance;
+			if (roll)
+			{
+				double bonus = amountToSteal * highRiskRewardBonus * 0.01;
+				victimCurrency.setCurrency(victimCurrency.currency() - amountToSteal);
+				authorCurrency.setCurrency(authorCurrency.currency() + amountToSteal + bonus);
+				QString num = QString::number(bonus);
+
+				QString stealAmount = QString::number(amountToSteal);
+				QString output = QString(
+					":man_detective: **Steal success!** :man_detective:\n"
+					"*%1* has discreetly stolen **`%2 %3`** from under *%4's* nose.\n"
+					"HighRiskHighReward event **BONUS**: `%5 %3`\n"
```

Comment:
You can just append this part of the message to the `outputString` or whatever after the main portion of the steal logic is done.